src/oci_image/images.js

```diff
@@ -0,0 +1,316 @@
+				errors.push(`${name} part '${value}' doesn't match allowed pattern '${pattern.source}'`);
+			}
+		}
+
+		if (errors.length > 0) {
+			const message = `Given Docker name '${this.getFullName()}' is invalid:\n` +
+				errors.map(error => `   * ${error}`).join('\n') +
+				'\nSee http://bit.ly/docker_image_fmt for more details';
+			throw new Error(message);
+		}
+	}
+
+	/**
+	* @param {string} rest
+	*/
+	parseComponentsBeforeTag(rest) {
+		const parts = rest.split(/\s*\/\s*/);
+		if (parts.length === 1) {
+			this.registry = null;
+			this.user = null;
+			this.repository = parts[0];
+		} else if (parts.length >= 2) {
+			if (this.isRegistry(parts[0])) {
+				this.registry = parts[0];
+				if (parts.length > 2) {
+					this.user = parts[1];
+					this.repository = this.joinTail(parts);
+				} else {
+					this.user = null;
+					this.repository = parts[1];
+				}
+			} else {
+				this.registry = null;
+				this.user = parts[0];
+				this.repository = rest;
+			}
+		}
+	}
+}
+
+export class ImageRef {
+	static OCI_TYPE = "oci";
+	static REPOSITORY_QUALIFIER = "repository_url";
+	static TAG_QUALIFIER = "tag";
+	static ARCH_QUALIFIER = "arch";
+	static OS_QUALIFIER = "os";
+	static VARIANT_QUALIFIER = "variant";
+
+	/** @type {Image} */
+	image;
+	/** @type {Platform} */
+	platform;
+
+	/**
+	 * @param {string} image
+	 * @param {string} [platform]
+	 */
+	constructor(image, platform) {
+		this.image = new Image(image);
+
+		if (platform != null) {
+			this.platform = Platform.fromString(platform);
+		}
+
+		this.checkImageDigest();
+	}
+
+	/**
+	 * @private
+	 */
+	checkImageDigest() {
+		if (this.image.digest == null) {
+			try {
+				const digests = getImageDigests(this);
+				if (digests.size === 0) {
+					throw new Error("Failed to get any image digest");
+				}
+				if (digests.size === 1 && digests[Platform.EMPTY.toString()]) {
+					this.image.digest = digests[Platform.EMPTY.toString()];
+				} else {
+					if (this.platform == null) {
+						this.platform = getImagePlatform();
+					}
+					if (this.platform == null) {
+						throw new Error(`Failed to get image platform for image digest`);
+					}
+					if (!digests[this.platform.toString()]) {
+						throw new Error(`Failed to get image digest for platform ${this.platform}`);
+					}
+					this.image.digest = digests[this.platform.toString()];
+				}
+			} catch (ex) {
+				throw new Error("Failed to get image digest", { cause: ex });
+			}
+		}
+	}
+
+	/**
+	 * @returns {PackageURL}
+	 * @throws {Error}
+	 * @see https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#oci
+	 */
+	getPackageURL() {
+		/** @type {Object.<string, string>} */
+		const qualifiers = {};
+		const repositoryUrl = this.image.getNameWithoutTag();
+		const simpleName = this.image.getSimpleName();
+
+		if (repositoryUrl != null && repositoryUrl.toLowerCase() !== simpleName.toLowerCase()) {
+			qualifiers[ImageRef.REPOSITORY_QUALIFIER, repositoryUrl.toLowerCase()];
```

Comment:
I think you are trying to assign the value here:
```suggestion
			qualifiers[ImageRef.REPOSITORY_QUALIFIER] = repositoryUrl.toLowerCase();
```